backend/app/models/launched_study.rb

```diff
@@ -11,8 +11,14 @@ def completed?
     completed_at.present?
   end
 
+  def retakeable?
+    # Timestamps are in microseconds - consider 0.1 sec window as the "same"
+    completed_at - opted_out_at < 0.1
+  end
+
   def completed!(consent: true)
-    update!(completed_at: Time.now, consent_granted: consent)
+    update!(completed_at: Time.now, consent_granted: consent,
```

Comment:
I suppose the timestamps would be the same if I wrapped this in a transaction, or fetched it into a variable and used that for both - yup,  I think that's what I should do, actually.